diff --git a/CMakeLists.txt b/CMakeLists.txt
index a59fd818e3..9f1bca3060 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3672,10 +3672,10 @@ if (gRPC_INSTALL)
   )
 endif()
 
-if (gRPC_BUILD_TESTS)
+if (gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
 
 if (gRPC_BUILD_CODEGEN)
-add_library(grpc++_proto_reflection_desc_db
+add_library(grpc++_proto_reflection_desc_db STATIC
   test/cpp/util/proto_reflection_descriptor_database.cc
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/reflection/v1alpha/reflection.pb.cc
   ${_gRPC_PROTO_GENS_DIR}/src/proto/grpc/reflection/v1alpha/reflection.grpc.pb.cc
@@ -3735,7 +3735,7 @@ foreach(_hdr
 endforeach()
 endif (gRPC_BUILD_CODEGEN)
 
-endif (gRPC_BUILD_TESTS)
+endif (gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
 
 if (gRPC_BUILD_CODEGEN)
 add_library(grpc++_reflection
@@ -3804,9 +3804,9 @@ if (gRPC_INSTALL)
   )
 endif()
 
-if (gRPC_BUILD_TESTS)
+if (gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
 
-add_library(grpc++_test_config
+add_library(grpc++_test_config STATIC
   test/cpp/util/test_config_cc.cc
 )
 
@@ -3845,8 +3845,7 @@ target_link_libraries(grpc++_test_config
   ${_gRPC_ALLTARGETS_LIBRARIES}
 )
 
-
-endif (gRPC_BUILD_TESTS)
+endif (gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
 if (gRPC_BUILD_TESTS)
 
 if (gRPC_BUILD_CODEGEN)
@@ -4558,10 +4557,10 @@ target_link_libraries(grpc_benchmark
 
 
 endif (gRPC_BUILD_TESTS)
-if (gRPC_BUILD_TESTS)
+if (gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
 
 if (gRPC_BUILD_CODEGEN)
-add_library(grpc_cli_libs
+add_library(grpc_cli_libs STATIC
   test/cpp/util/cli_call.cc
   test/cpp/util/cli_credentials.cc
   test/cpp/util/grpc_tool.cc
@@ -4626,7 +4625,7 @@ foreach(_hdr
 endforeach()
 endif (gRPC_BUILD_CODEGEN)
 
-endif (gRPC_BUILD_TESTS)
+endif (gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
 
 add_library(grpc_plugin_support
   src/compiler/cpp_generator.cc
@@ -11885,7 +11884,7 @@ target_link_libraries(grpc_alts_credentials_options_test
 )
 
 endif (gRPC_BUILD_TESTS)
-if (gRPC_BUILD_TESTS)
+if (gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
 
 add_executable(grpc_cli
   test/cpp/util/grpc_cli.cc
@@ -11924,7 +11923,15 @@ target_link_libraries(grpc_cli
   ${_gRPC_GFLAGS_LIBRARIES}
 )
 
-endif (gRPC_BUILD_TESTS)
+if (gRPC_INSTALL)
+  install(TARGETS grpc_cli EXPORT gRPCTargets
+    RUNTIME DESTINATION ${gRPC_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${gRPC_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${gRPC_INSTALL_LIBDIR}
+  )
+endif()
+
+endif (gRPC_BUILD_TESTS OR gRPC_BUILD_CLI)
 if (gRPC_BUILD_CODEGEN)
 
 add_executable(grpc_cpp_plugin
