From 9a50fe32f254db7fb61137833c7d41aa8a0ec4a9 Mon Sep 17 00:00:00 2001
From: Rubens Figueiredo <rubens.figueiredo@bisdn.de>
Date: Mon, 22 Mar 2021 18:49:54 +0100
Subject: [PATCH] add for baseboxd in service file

---
 keepalived/keepalived.service.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/keepalived/keepalived.service.in b/keepalived/keepalived.service.in
index 75bf0e50..56255f2f 100644
--- a/keepalived/keepalived.service.in
+++ b/keepalived/keepalived.service.in
@@ -1,6 +1,6 @@
 [Unit]
 Description=LVS and VRRP High Availability Monitor
-After=network-online.target syslog.target @SNMP_SERVICE@
+After=network-online.target syslog.target @SNMP_SERVICE@ baseboxd.service
 Wants=network-online.target @SNMP_SERVICE@
 Documentation=man:keepalived(8)
 Documentation=man:keepalived.conf(5)
-- 
2.29.2
