From c870e98db440f80a0f3c73136c2dfd9e048700ce Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Thu, 28 Jan 2021 14:35:30 +0100
Subject: [PATCH] bridge: print readable vlan state

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>
---
 bridge/vlan.c | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/bridge/vlan.c b/bridge/vlan.c
index 7ba8468f..846ef7aa 100644
--- a/bridge/vlan.c
+++ b/bridge/vlan.c
@@ -522,6 +522,25 @@ static int print_vlan_stats(struct nlmsghdr *n, void *arg)
 	return 0;
 }
 
+static const char *port_states[] = {
+	[BR_STATE_DISABLED] = "disabled",
+	[BR_STATE_LISTENING] = "listening",
+	[BR_STATE_LEARNING] = "learning",
+	[BR_STATE_FORWARDING] = "forwarding",
+	[BR_STATE_BLOCKING] = "blocking",
+};
+
+
+static void print_portstate(__u8 state)
+{
+	if (state <= BR_STATE_BLOCKING)
+		print_string(PRINT_ANY, "state",
+			     "%s", port_states[state]);
+	else
+		print_uint(PRINT_ANY, "state",
+			     "unknown (%d) ", state);
+}
+
 static int print_vlan_options(struct nlmsghdr *n, void *arg)
 {
 	struct br_vlan_msg *br_msg = NLMSG_DATA(n);
@@ -577,7 +596,7 @@ static int print_vlan_options(struct nlmsghdr *n, void *arg)
 			fprintf(stderr, "BUG: vlan range too wide, %u\n",
 				width);
 		}
-		print_uint(PRINT_ANY, "stp_state", "%u", state);
+		print_portstate(state);
 		close_json_object();
 
 		print_nl();
-- 
2.30.0

