From aaec0f86cf7f73f7f6e98d33c0e141f5f43b4692 Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Fri, 9 Oct 2020 10:09:03 +0200
Subject: [PATCH 1/2] route/link: print slave line

---
 lib/route/link.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/route/link.c b/lib/route/link.c
index d73c5a4..45a6a22 100644
--- a/lib/route/link.c
+++ b/lib/route/link.c
@@ -937,6 +937,9 @@ static void link_dump_line(struct nl_object *obj, struct nl_dump_params *p)
 	if (link->l_info_ops && link->l_info_ops->io_dump[NL_DUMP_LINE])
 		link->l_info_ops->io_dump[NL_DUMP_LINE](link, p);
 
+	if (link->l_info_slave_ops && link->l_info_slave_ops->io_slave_dump[NL_DUMP_LINE])
+		link->l_info_slave_ops->io_slave_dump[NL_DUMP_LINE](link, p);
+
 	do_foreach_af(link, af_dump_line, p);
 
 	nl_dump(p, "\n");
-- 
2.28.0

