From 49987eab893a29933d13492a100b6441855d78f7 Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Mon, 12 Oct 2020 12:17:40 +0200
Subject: [PATCH 4/4] route/link/bonding: add missing compare to slave_ops

---
 lib/route/link/bonding.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/lib/route/link/bonding.c b/lib/route/link/bonding.c
index 0293510..31000d5 100644
--- a/lib/route/link/bonding.c
+++ b/lib/route/link/bonding.c
@@ -270,6 +270,22 @@ static int bond_compare(struct rtnl_link *link_a, struct rtnl_link *link_b,
 	return diff;
 }
 
+static int bond_slave_compare(struct rtnl_link *link_a, struct rtnl_link *link_b,
+			 int flags)
+{
+	struct bond_slave_info *a = link_a->l_info_slave;
+	struct bond_slave_info *b = link_b->l_info_slave;
+	int diff = 0;
+	uint32_t attrs = flags & LOOSE_COMPARISON ? b->ce_mask : ~0;
+
+#define BOND_SLAVE_DIFF(ATTR, EXPR) ATTR_DIFF(attrs, BOND_SLAVE_ATTR_##ATTR, a, b, EXPR)
+
+	diff |= BOND_SLAVE_DIFF(STATE, a->bsi_state != b->bsi_state);
+#undef BONDSLAVE__DIFF
+
+	return diff;
+}
+
 /**
  * Allocate link object of type bond
  *
@@ -480,6 +496,7 @@ static struct rtnl_link_info_ops bonding_info_ops = {
 	.io_slave_dump = {
 	    [NL_DUMP_LINE]	= bond_slave_dump_line,
 	},
+	.io_slave_compare	= bond_slave_compare,
 };
 
 /** @cond SKIP */
-- 
2.28.0

