From 0474e570fff610eb70adb62afb6ab63a33412096 Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Tue, 1 Mar 2022 12:59:50 +0100
Subject: [PATCH 09/10] route/route_obj: treat each IPv6 link-local route as
 different

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>
---
 lib/route/route_obj.c | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/lib/route/route_obj.c b/lib/route/route_obj.c
index 6d36f80d2e13..93f3d87b9c96 100644
--- a/lib/route/route_obj.c
+++ b/lib/route/route_obj.c
@@ -389,6 +389,35 @@ static uint32_t route_id_attrs_get(struct nl_object *obj)
 	if (route->rt_family == AF_MPLS)
 		rv &= ~ROUTE_ATTR_PRIO;
 
+	/* pretend link-local IPv6 routes are multipath so that different
+	 * nexthops (OIF) will be treated as different routes */
+	if (route->rt_dst && nl_addr_get_family(route->rt_dst) == AF_INET6) {
+		uint8_t *addr = nl_addr_get_binary_addr(route->rt_dst);
+		unsigned int prefixlen = nl_addr_get_prefixlen(route->rt_dst);
+
+		switch (route->rt_type) {
+		case RTN_UNICAST:
+			if (prefixlen == 64 &&
+			    addr[0] == 0xfe && (addr[1] & 0xc0) == 0x80)
+				/* link-local unicast */
+				rv |= ROUTE_ATTR_MULTIPATH;
+			break;
+		case RTN_ANYCAST:
+			if (prefixlen == 128 &&
+			    addr[0] == 0xfe && (addr[1] & 0xc0) == 0x80)
+				/* link-local anycast */
+				rv |= ROUTE_ATTR_MULTIPATH;
+			break;
+		case RTN_MULTICAST:
+			if (prefixlen == 8 && addr[0] == 0xff)
+				/* multicast */
+				rv |= ROUTE_ATTR_MULTIPATH;
+			break;
+		default:
+			break;
+		}
+	}
+
 	return rv;
 }
 
-- 
2.45.1

